src/app/auth/services/auth.service.ts

```diff
@@ -1,32 +1,111 @@
+  providedIn: 'root',
 })
 export class AuthService {
-  login(user: any) {
-    // replace 'any' with the required interface
-    // Add your code here
+  private isAuthorized$$ = new BehaviorSubject<boolean>(false);
+  public isAuthorized$: Observable<boolean> = this.isAuthorized$$.asObservable();
+
+  constructor(
+    private http: HttpClient,
+    private sessionStorage: SessionStorageService,
+    private userStore: UserStoreService
+  ) {}
+
+  login(user: LoginUser): Observable<APIResult> {
+    return this.http.post<LoginResponse>(this.getLoginUrl(), user).pipe(
+      map(response => {
+        const token = response?.result;
+        if (token) {
+          this.sessionStorage.setToken(token);
+          this.isAuthorized$$.next(true);
+          if (response.user.email === ADMIN_EMAIL) {
+            this.userStore.isAdmin = true;
+            this.userStore.userName = 'Admin';
+          } else {
+            this.userStore.isAdmin = false;
+            this.userStore.userName = response.user.name;
+          }
+          return { result: true, email: user.email };
+        } else {
+          this.isAuthorized$$.next(false);
+          return {
+            result: false,
+            error: 'Token is missing from server',
+          };
+        }
+      }),
+      catchError(err => {
+        throw 'Login failed: ' + JSON.stringify(err.error.result);
+      })
+    );
   }
 
-  logout() {
-    // Add your code here
+  logout(): Observable<APIResult> {
+    const token = this.sessionStorage.getToken();
+    const headers = new HttpHeaders({
+      Authorization: `${token}`,
+    });
+
+    return this.http.delete<LogoutResponse>(this.getLogoutUrl(), { headers }).pipe(
+      map(() => {
+        this.sessionStorage.deleteToken();
+        this.isAuthorized$$.next(false);
+        this.userStore.userName = '';
+        this.userStore.isAdmin = false;
+        return {
+          result: true,
+        };
+      }),
+      catchError(error => {
+        return of({
+          result: false,
+          error: error?.message || 'Unknown error',
+        });
+      })
+    );
   }
 
-  register(user: any) {
-    // replace 'any' with the required interface
-    // Add your code here
+  register(user: User): Observable<APIResult> {
+    return this.http.post<ResgistrationResponse>(this.getRegistrationUrl(), user).pipe(
+      map((response: ResgistrationResponse) => {
+        return {
+          result: response.successful,
+          error: response.errors[0],
+        };
+      }),
+      catchError(error => {
+        return of({
+          result: false,
+          error: error?.message || 'Unknown error',
+        });
+      })
+    );
   }
 
   get isAuthorised() {
-    // Add your code here. Get isAuthorized$$ value
+    return this.isAuthorized$$.getValue();
   }
 
   set isAuthorised(value: boolean) {
-    // Add your code here. Change isAuthorized$$ value
+    this.isAuthorized$$.next(value);
   }
 
   getLoginUrl() {
```

Comment:
This is a classic example of managing environments in an Angular application. Suppose you are working on the development locally, but it will be deployed to some testing and a production environment later.
Angular has a designated place to manage it - under src/environments folder. You can add your URLs, env variables there individually. Also check the config in .angular file. Read more here: https://dev.to/remshams/replacing-configuration-files-in-angular-builds-based-on-build-targets-28jm

Reply:
Sounds better! : ) Thank you!